File: app/src/main/java/com/lewis/liveclient/view/BaseActivity.java
package com.lewis.liveclient.view;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

import com.lewis.liveclient.Annotation.AndroidAnnotation;

/**
 * Created by Lewis on 2017/11/26.
 *
 */

public abstract class BaseActivity extends AppCompatActivity {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        init();
        try {
            AndroidAnnotation.init(this);
        } catch (ClassNotFoundException e) {
            e.printStackTrace();
        } catch (IllegalAccessException e) {
            e.printStackTrace();
        }
    }

    /**
     * setContentView(R.layout.activity_my)在此方法中进行
     */
    protected abstract void init();
}

File: app/src/main/java/com/lewis/liveclient/view/BaseActivity.kt
package com.lewis.liveclient.view

import android.Manifest
import android.content.pm.PackageManager
import android.os.Build
import android.os.Bundle
import android.os.PersistableBundle
import android.support.v4.app.ActivityCompat
import android.support.v7.app.AppCompatActivity

/**
 * Created by lewis on 18-2-20.
 *
 */
abstract class BaseActivity : AppCompatActivity() {

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    //Todo
    requestPermissions()
    init()
  }

  abstract fun init()

  fun requestPermissions() {
    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
      //camera
      if (ActivityCompat.checkSelfPermission(this, Manifest.permission.CAMERA)
          !=  PackageManager.PERMISSION_GRANTED) {
        ActivityCompat.requestPermissions(this
            , arrayOf(Manifest.permission.CAMERA), PackageManager.PERMISSION_GRANTED)
      }
      //READ_EXTERNAL_STORAGE
      if (ActivityCompat.checkSelfPermission(this, Manifest.permission.READ_EXTERNAL_STORAGE)
          !=  PackageManager.PERMISSION_GRANTED) {
        ActivityCompat.requestPermissions(this
            , arrayOf(Manifest.permission.READ_EXTERNAL_STORAGE), PackageManager.PERMISSION_GRANTED)
      }
      //WRITE_EXTERNAL_STORAGE
      if (ActivityCompat.checkSelfPermission(this, Manifest.permission.WRITE_EXTERNAL_STORAGE)
          !=  PackageManager.PERMISSION_GRANTED) {
        ActivityCompat.requestPermissions(this
            , arrayOf(Manifest.permission.WRITE_EXTERNAL_STORAGE), PackageManager.PERMISSION_GRANTED)
      }
    }
  }
}